src/backup-all-indexes.sh

```diff
@@ -88,18 +152,74 @@ curl -w "\n" -sS -XPUT ${REPOSITORY_URL} -d "{
   }
 }"
 
+#define array to store results
+arr=()
 CUTOFF_DATE=$(date --date="${MAX_DAYS_TO_KEEP} days ago" +"%Y.%m.%d")
+
 echo "$(now) Archiving all indexes with logs before ${CUTOFF_DATE}."
-SUBSTITUTION='s/.*\(logstash-[0-9\.]\{10\}\).*/\1/'
-for index_name in $(curl -sS ${DATABASE_URL}/_cat/indices | grep logstash- | sed $SUBSTITUTION | sort); do
-  if [[ "${index_name:9}" < "${CUTOFF_DATE}" ]]; then
-    echo "$(now): Ensuring ${index_name} is archived..."
-      backup_index ${index_name}
-      if [ $? -eq 0 ]; then
-        echo "$(now): ${index_name} archived."
-      else
-        echo "$(now): ${index_name} archival failed."
-      fi
-  fi
+
+# itterate array of indices
+for index_nm in ${INDEX_ARRAY[@]}; do
+
+  #set counters for ar - archived index, bk - backuped and fl - failed
+  ar=0
+  bk=0
+  fl=0
+
+
+    # regexp to catch index with date from string
+    SUBSTITUTION="s/.*\($index_nm-[0-9\.]\{10\}\).*/\1/"
+    SUBSTITUTION_date="s/.*([0-9]{4}\.[0-9]{2}\.[0-9]{2}).*/\1/"
+       # intterate index strings
+       for index in $(curl -sS ${DATABASE_URL}/_cat/indices | grep $index_nm- | sed $SUBSTITUTION | sort); do
+
+          # debug information
+          echo "$index"
+          CURR_DATE=`echo $index | sed -re $SUBSTITUTION_date`
+          # if date less then cutoff date, archive index to s3 and delete from elasticsearch
+          if [[ "$CURR_DATE" < "${CUTOFF_DATE}" ]]; then
+
+ 
+            echo "$(now): Ensuring ${index} is archived..."
+
+            archive_index ${index}
+ 
+            if [ $? -eq 0 ]; then
+              echo "$(now): ${index} archived."
+              let ar++
+            else
+              echo "$(now): ${index} archival failed."
+               let fl++ 
+           fi
+         # if less then cuttoff date, just backup it to s3 without deletion
+         else 
```

Comment:
All the logic above just checks for "*any* snapshot exists" to mean "this is safe to delete" (because that's how this has worked so far). However, if you're now snapshotting indices regardless of their creation date, you might end up creating a snapshot for *today*, which will be incomplete.

Unfortunately, since that means a snapshot exists, it seems that no final snapshot will be created before finally deleting the index.

Am I missing something here?